stock_batch_picking/models/stock_picking.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# @2016 Cyril Gaudin, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    batch_picking_id = fields.Many2one(
+        comodel_name='stock.batch.picking',
+        string='Batch',
+        copy=False,
+        domain="[('state', '=', 'draft')]",
+        help='In which picking dispatch this move will be processed.'
+    )
+
+    @api.multi
+    def action_cancel(self):
+        """In addition to what the method in the parent class does,
+        cancel the batches for which all pickings are cancelled
+        """
+        result = super(StockPicking, self).action_cancel()
+        self.mapped('batch_picking_id').verify_state()
+
+        return result
+
+    @api.multi
+    def action_assign(self):
+        """In addition to what the method in the parent class does,
+        Changed batches states to assigned if all picking are assigned.
+        """
+        result = super(StockPicking, self).action_assign()
+        self.mapped('batch_picking_id').verify_state('assigned')
+
+        return result
+
+    @api.multi
+    def do_transfer(self):
+        """In addition to what the method in the parent class does,
+        Changed batches states to done if all picking are done.
+        """
+        result = super(StockPicking, self).do_transfer()
+        self.mapped('batch_picking_id').verify_state()
+
+        return result
+
+    def force_transfer(self, force_qties=True):
```

Comment:
IMO `force_qties` is going to be really hard to remember, as `qties` is not a real abbreviation - `force_qty` would be better